### PR TITLE
types: fix uint64 overflow bug in `ConvertJSONToFloat`

### PR DESCRIPTION
**Pull request by H-ZeX**
### What problem does this PR solve? 

In the origin version `ConvertIntToUint(sc, j.GetInt64(), IntergerUnsignedUpperBound(mysql.TypeLonglong), mysql.TypeLonglong)` , when `j.GetInt64()` return neg number such as `1<<63`, then `ConvertIntToUint` may treat it as overflow, this is a bug.

### What is changed and how it works?

Replace `j.GetInt64` by `j.GetUint64`

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: types/convert.go [hunk 1]
```diff
@@ -553,8 +553,7 @@ func ConvertJSONToFloat(sc *stmtctx.StatementContext, j json.BinaryJSON) (float6
 	case json.TypeCodeInt64:
 		return float64(j.GetInt64()), nil
 	case json.TypeCodeUint64:
-		u, err := ConvertIntToUint(sc, j.GetInt64(), IntergerUnsignedUpperBound(mysql.TypeLonglong), mysql.TypeLonglong)
```
Comment (winkyao) [inline types/convert.go]: Shall we consider `ShouldClipToZero`?

Comment (H-ZeX) [reply types/convert.go]: I think we need to consider this.
we extract a uint from the json, it must not < 0.
```go
// GetInt64 gets the int64 value.
func (bj BinaryJSON) GetInt64() int64 {
	return int64(endian.Uint64(bj.Value))
}

// GetUint64 gets the uint64 value.
func (bj BinaryJSON) GetUint64() uint64 {
	return endian.Uint64(bj.Value)
}
```

Comment (winkyao) [reply types/convert.go]: need or needn't?

Comment (H-ZeX) [reply types/convert.go]: needn't

**Comment by H-ZeX** [comment]
@winkyao

File: expression/integration_test.go [hunk 2]
```diff
@@ -2172,10 +2172,14 @@ func (s *testIntegrationSuite) TestBuiltin(c *C) {
 	result.Check(testkit.Rows("<nil>"))
 	result = tk.MustQuery(`select cast(cast('2017-01-01 01:01:11.12' as date) as datetime(2));`)
 	result.Check(testkit.Rows("2017-01-01 00:00:00.00"))
-
 	result = tk.MustQuery(`select cast(20170118.999 as datetime);`)
 	result.Check(testkit.Rows("2017-01-18 00:00:00"))
 
+	tk.MustExec(`create table tb5(a bigint(64) unsigned, b double);`)
```
Comment (zz-jason) [inline expression/integration_test.go]: why not directly create a json column?

Comment (H-ZeX) [reply expression/integration_test.go]: If directly creat json, then I can not get a uint field(https://github.com/pingcap/tidb/issues/11386#issuecomment-514594521).

Comment (zz-jason) [reply expression/integration_test.go]: how about casting '9223372036854775808' to uint then cast to json?

**Pull request merged**